component/attr_cache/attr_cache.go

```diff
@@ -455,34 +410,27 @@ func (ac *AttrCache) ReadDir(options internal.ReadDirOptions) (pathList []*inter
 
 // merge results from our cache into pathMap
 func (ac *AttrCache) addDirsNotInCloudToListing(listPath string, pathList []*internal.ObjAttr) ([]*internal.ObjAttr, int) {
-	prefix := dirToPrefix(listPath)
-	if prefix == "/" {
-		prefix = ""
-	}
 	numAdded := 0
-	ac.cacheLock.RLock()
-	for key, value := range ac.cacheMap {
-		// ignore invalid and deleted items
-		if !value.valid() || !value.exists() {
-			continue
-		}
-		// the only entries missing are the directories with no objects
-		if !value.attr.IsDir() || value.isInCloud() {
-			continue
-		}
-		// look for matches
-		if strings.HasPrefix(key, prefix) {
-			// exclude entries in subdirectories
-			pathInsideDirectory := strings.TrimPrefix(key, prefix)
-			if strings.Contains(pathInsideDirectory, "/") {
-				continue
-			}
 
-			pathList = append(pathList, value.attr)
-			numAdded++
+	nonCloudItem, err := ac.cacheMap.get(listPath)
+
+	if err != nil {
+		log.Err("could not find the attr cached item: ", err)
+		return nil, 0
```

Comment:
```suggestion
		return pathList, 0
```
We have to return the merged list. So if we found nothing, we still need to return what we were given.